src/NumericalAlgorithms/Spectral/SwshTransformJob.hpp

```diff
@@ -0,0 +1,406 @@
+ * - `Spin`: The spin weight for all of the fields to transform with this job.
+ * - `Representation`: The element of the `ComplexRepresentation` enum which
+ *   parameterizes how the data passed to libsharp will be represented. Two
+ *   options are available:
+ *  - `ComplexRepresentation:Interleaved`: indicates that the real and imaginary
+ *    parts of the collocation values will be passed to libsharp as pointers
+ *    into existing complex data structures, minimizing copies, but maintaining
+ *    a stride of 2 for 'adjacent' real or imaginary values.
+ *  - `ComplexRepresentation::RealsThenImags`: indicates that the real and
+ *    imaginary parts of the collocation values will be passed to libsharp as
+ *    separate contiguous blocks. At current, this introduces both allocations
+ *    and copies.  **optimization note** In the future most of the allocations
+ *    can be aggregated to calling code, which would pass in buffers by
+ *    `not_null` pointers.
+ *
+ *  For performance-sensitive code, both options should be tested, as each
+ *  strategy has trade-offs.
+ * - `TagList`: A `tmpl::list` of Tags to be forward and/or backward transformed
+ *   with the `TransformJob`.
+ *
+ * \note The signs obtained from libsharp transformations must be handled
+ * carefully. (In particular, it does not use the sign convention you will find
+ * in [Wikipedia]
+ * (https://en.wikipedia.org/wiki/Spin-weighted_spherical_harmonics)).
+ * - libsharp deals only with the transformation of real values, so
+ *   transformation of complex values must be done in multiple components.
+ * - due to only transforming real components, it stores only the positive
+ *   \f$m\f$ modes (as the rest would be redundant). Therefore, the negative
+ *   \f$m\f$ modes must be inferred from conjugation and further sign changes.
+ * - libsharp only has the capability of transforming positive spin weighted
+ *   quantities. Therefore, additional steps are taken (involving further
+ *   conjugation of the provided data) in order to use those utilities on
+ *   negative spin-weighted quantities. The resulting modes have yet more sign
+ *   changes that must be taken into account.
+ *
+ * The decomposition resulting from the libsharp transform for spin \f$s\f$ and
+ * complex spin-weighted \f${}_s f\f$ can be represented mathematically as:
+ *
+ * \f{align*}{
+ * {}_s f(\theta, \phi) = \sum_{\ell = 0}^{l_\mathrm{max}} \Bigg\{&
+ * \left(\sum_{m = 0}^{\ell} a^\mathrm{sharp, real}_{l m} {}_s Y_{\ell
+ * m}^\mathrm{sharp, real}\right) + \left(\sum_{m=1}^{\ell}
+ * \left(a^\mathrm{sharp, real}_{l m}{}\right)^*
+ * {}_s Y_{\ell -m}^\mathrm{sharp, real}\right) \notag\\
+ * &+ i \left[\left(\sum_{m = 0}^{\ell} a^\mathrm{sharp, imag}_{l m} {}_s
+ * Y_{\ell m}^\mathrm{sharp, imag}\right) + \left(\sum_{m=1}^{\ell}
+ * \left(a^\mathrm{sharp, imag}_{l m}{}\right)^* {}_s Y_{\ell -m}^\mathrm{sharp,
+ * imag} \right)\right] \Bigg\},
+ * \f}
+ *
+ * where
+ *
+ * \f{align*}{
```

Comment:
Be consistent and pick either `\f$l\f$` or `\f$\ell\f$` everywhere in this dox comment. Both are currently used.